config/hooks/pre-commit

```diff
@@ -16,24 +17,25 @@ echo "Linting Python code..."
 echo "$staged_files" | grep -E --quiet ".+.py" && {
     staged_python_files=$(echo "$staged_files" | grep -E ".+.py")
     # Autoformat and lint Python code.
-    poetry --help &>/dev/null && [[ -d ".venv" ]] && {
+    poetry --quiet && [[ -d ".venv" ]] && {
         comma_delimited_filepaths=$(echo "$staged_python_files" | tr '\n' ',')
+        space_delimited_filepaths=$(echo "$staged_python_files" | tr '\n' ' ')
         echo "Autoformatting Python code..."
         poetry run invoke qa.autoformat --filepaths "${comma_delimited_filepaths%,}"
-        for file in $staged_python_files; do
-            # Check types with mypy first, since mypy is faster than pytype.
-            # https://github.com/google/pytype#usage
-            poetry run mypy "$file" || {
-                echo "
-                    ${BOLD}Examine the error above. Should it be fixed?${NORMAL}
-                    Pytype may give more helpful output:
-                        poetry run pytype --output .cache/pytype $file
-                "
-            }
-        done
-        for file in $staged_python_files; do
-            poetry run flake8 "$file" || linting_failed=true
-        done
+
+        # Check types with mypy first, since mypy is faster than pytype.
+        # https://github.com/google/pytype#usage
+        poetry run mypy "$space_delimited_filepaths" || {
+            echo "
+                ${BOLD}Examine the error above. Should it be fixed?${NORMAL}
+                Pytype may give more helpful output:
+                    poetry run pytype --output .cache/pytype $file
+            "
+        }
+
+
+        poetry run flake8 "$space_delimited_filepaths" || linting_failed=true
+
```

Comment:
I'm not sure if there was a specific reason to re-run these for each file, but since the checks run recursively through imports, it was taking forever to run on individual files. This should deduplicate that work.